lib/misc/firestore.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:latlong2/latlong.dart';
+import 'package:tg_proj/misc/auth.dart';
+
+class Firestore {
+  static final Firestore instance = Firestore._();
+
+  Firestore._();
+
+  final user = Auth.instance.currentUser;
+
+  Future<void> createDocOnRegister() async {
+    await FirebaseFirestore.instance.collection('users').doc(user!.uid).set({
+      'email': user!.email,
+      'streak': 0,
+      'registered_on': Timestamp.now(),
+      'total_distance': 0,
+      'challenges_completed': 0
+    });
+
+    await FirebaseFirestore.instance
+        .collection('users')
+        .doc(user!.uid)
+        .collection('challenge_history')
+        .doc('history')
+        .set({});
+  }
+
+  Future<int> getCountOfChallenges() async {
+    final docSnapshot = await FirebaseFirestore.instance
+        .collection('users')
+        .doc(user!.uid)
+        .get();
+    final challengesCompleted = docSnapshot.data()?['challenges_completed'];
+
+    return challengesCompleted;
+  }
+
+  Future<void> addChallengeToHistory(
+      double lat, double lng, int distance) async {
+    final int index = await getCountOfChallenges();
+    final usrData =
+        FirebaseFirestore.instance.collection('users').doc(user!.uid);
+
+    final docSnapshot = await usrData.get();
+
+    final totalDistance = docSnapshot.data()?['total_distance'];
```

Comment:
What happens if total_distance is not there? I suppose it returns `null`
You probably want to return 0 if that happens